classes/issue/IssueDAO.inc.php

```diff
@@ -174,6 +174,51 @@ function getPublishedIssuesByNumber($journalId, $volume = null, $number = null,
 	}
 
 	/**
+	 * Retrieve Issues by identification
+	 * @param $journalId int
+	 * @param $volume int
+	 * @param $number string
+	 * @param $year int
+	 * @param $titles array
+	 * @return DAOResultFactory
+	 */
+	function getIssuesByIdentification($journalId, $volume = null, $number = null, $year = null, $titles = array()) {
+		$params = array((int) $journalId);
+
+		$sqlVol = $sqlNum = $sqlYear = '';
+		if ($volume !== null) {
+			$sqlVol = ' AND i.volume = ?';
+			$params[] = (int) $volume;
+		}
+		if ($number !== null) {
+			$sqlNum = ' AND i.number = ?';
+			$params[] = $number;
+		}
+		if ($year !== null) {
+			$sqlYear = ' AND i.year = ?';
+			$params[] = (int) $year;
+		}
+
+		$i = 1;
+		$sqlTitleJoin = $sqlTitleWhere = '';
+		foreach ($titles as $title) {
+			$sqlTitleJoin .= ' LEFT JOIN issue_settings iss' .$i .' ON (i.issue_id = iss' .$i .'.issue_id)';
```

Comment:
If you use a `JOIN` rather than a `LEFT JOIN`, you can put all the conditions into the `ON (...)` clause. `LEFT JOIN` is only needed when you want the match to be "optional".